src/azul/plugins/metadata/hca/transform.py

```diff
@@ -841,39 +846,70 @@ def _sequencing_input(self, sequencing_input: api.Biomaterial) -> MutableJSON:
-        # possible from any entities that can be a sample. This is done to
-        # have consistency of fields between various sample inner entities
-        # to allow Elasticsearch to search and sort against these entities.
-        sample_ = dict.fromkeys(ChainMap(
-            self._cell_line_types(),
-            self._organoid_types(),
-            self._specimen_types()
-        ).keys())
-        entity_type, entity_dict = (
-            'cell_lines', self._cell_line(sample)
-        ) if isinstance(sample, api.CellLine) else (
-            'organoids', self._organoid(sample)
-        ) if isinstance(sample, api.Organoid) else (
-            'specimens', self._specimen(sample)
-        ) if isinstance(sample, api.SpecimenFromOrganism) else (
-            require(False, sample), None
-        )
-        sample_.update(entity_dict)
-        sample_['entity_type'] = entity_type
-        assert hasattr(sample, 'organ') != hasattr(sample, 'model_organ')
-        sample_['effective_organ'] = sample.organ if hasattr(sample, 'organ') else sample.model_organ
-        assert sample_['document_id'] == str(sample.document_id)
-        assert sample_['biomaterial_id'] == sample.biomaterial_id
-        return sample_
+    def _samples(self, samples: List[api.Biomaterial]) -> Mapping[str, JSONs]:
+        """
+        Returns inner entity values with all given samples represented both in a
+        'samples' inner entity and a type-specific 'sample_{entity_type}'.
+        The 'samples' inner entity is a minified polymorphic field containing
+        the few properties common to all samples. This allows facets on these
+        properties regardless of the sample entity type.
+        """
+        inner_entities = defaultdict(list)
+        inner_entities['samples'] = []
+        entity_types = [
+            ('cell_lines', api.CellLine, self._cell_line),
+            ('organoids', api.Organoid, self._organoid),
+            ('specimens', api.SpecimenFromOrganism, self._specimen),
+        ]
+        for entity_type, metadata_class, entity_fn in entity_types:
+            entity_dicts = [entity_fn(s) for s in samples if isinstance(s, metadata_class)]
+            inner_entities[f'sample_{entity_type}'].extend(entity_dicts)
+            for entity in entity_dicts:
+                inner_entities['samples'].append(
+                    {
+                        'document_id': entity['document_id'],
+                        'biomaterial_id': entity['biomaterial_id'],
+                        'entity_type': entity_type,
+                        **(
+                            {
+                                'organ': None,
+                                'organ_part': None,
+                                'model_organ': entity['model_organ'],
+                                'model_organ_part': None,
+                                'effective_organ': entity['model_organ'],
+                            } if entity_type == 'cell_lines' else {}
```

Comment:
This is nested the wrong way. Only one of the conditionals can ever be true but all of them are tested. You should chain the `else` branches so no more tests are made once one tests returned true. 

Alternatively, and I think that's more elegant, define three nested functions and have your entity_types thingy include  references to those functions.